lib/galaxy/datatypes/flow.py

```diff
@@ -0,0 +1,51 @@
+"""
+Flow analysis datatypes.
+"""
+
+import logging
+
+from galaxy.datatypes.binary import Binary
+from galaxy.datatypes.sniff import build_sniff_from_prefix
+from . import data
+
+log = logging.getLogger(__name__)
+
+
+@build_sniff_from_prefix
+class FCS(Binary):
+    """Class describing an FCS binary file"""
+    file_ext = "fcs"
+
+    def set_peek(self, dataset, is_multi_byte=False):
+        if not dataset.dataset.purged:
+            dataset.peek = "Binary FCS file"
+            dataset.blurb = data.nice_size(dataset.get_size())
+        else:
+            dataset.peek = 'file does not exist'
+            dataset.blurb = 'file purged from disk'
+
+    def display_peek(self, dataset):
+        try:
+            return dataset.peek
+        except Exception:
+            return "Binary FCS file"
+
+    def sniff_prefix(self, file_prefix):
+        """
+        Checking if the file is in FCS format. Should read FCS2.0, FCS3.0
+        and FCS3.1
+
+        Based on flowcore:
+        https://github.com/RGLab/flowCore/blob/27141b792ad65ae8bd0aeeef26e757c39cdaefe7/R/IO.R#L667
+        """
+        content = file_prefix.contents_header_bytes[:42].decode()
+        version = content[:6]
+        assert version in ["FCS2.0", "FCS3.0", "FCS3.1"]
+        assert content[6:10] == '    '
```

Comment:
It's better to return `False` instead of using `assert`s, which can be disabled via `python -O`.

Reply:
Wow, I did not know that and this seems fully insane 😆

Reply:
@pcm32 can you address this

Reply:
Ok, done now! Hopefully I didn't introduce any linting issues.